contracts/admin/src/common/mod.rs

```diff
@@ -0,0 +1,124 @@
+use std::error::Error;
+
+use serde::{Deserialize, Serialize};
+
+use platform::batch::Batch;
+use sdk::{
+    cosmwasm_std::{Addr, Binary, QuerierWrapper, WasmMsg},
+    schemars::{self, JsonSchema},
+};
+
+use crate::{
+    common::type_defs::MaybeMigrateContract,
+    error::{ContractError, Never},
+};
+
+pub(crate) mod type_defs;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CodeIdWithMigrateMsg<M> {
+    pub code_id: u64,
+    pub migrate_msg: M,
+}
+
+pub(crate) trait ValidateAddresses {
+    fn validate(&self, querier: &QuerierWrapper<'_>) -> Result<(), ContractError>;
+}
+
+pub(crate) trait MigrateContracts {
+    type GatSelf<T>;
+
+    type Error: Into<ContractError> + Error;
+
+    fn migrate(
+        self,
+        migration_msgs: Self::GatSelf<MaybeMigrateContract>,
+    ) -> Result<Batch, Self::Error>;
```

Comment:
returning Batch is enough ... which makes the `type Error` useless

Reply:
Intention was to not have to change the interface when, inevitably, we go back to having the variant of missing symbols. I'll remove it if you think it's not necessary.

Reply:
yes, please